core/offchain/src/api.rs

```diff
@@ -608,6 +615,23 @@ mod tests {
 		assert_eq!(timestamp.unix_millis(), d);
 	}
 
+	#[test]
+	fn should_sleep() {
+		let mut api = offchain_api().0;
+
+		// Arrange.
+		let now = api.timestamp();
+		let delta = Duration::from_millis(100);
+		let t1 = now.add(delta);
+
+		// Act.
+		api.sleep_until(t1);
+		let t2 = api.timestamp();
+		
+		// Assert.
+		assert_eq!(t2.unix_millis() - now.unix_millis(), 100);
```

Comment:
I'd rely on being so specific here. It's fine if the different is at least 100 millis, but it might be longer.

Reply:
Yeah, after reading the docs for `thread::sleep` it seems the diff would be at least the sleep duration but could be more than that (because of sys calls).

Asserting it now accordingly:
```
new_now.unix_millis() - 100 >= now.unix_millis()
```